discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+                    }],
+                    'serviceNetwork': [service_cidr]
+                },
+                'provisionRequirements': {
+                    'controlPlaneAgents': control_plane_agents,
+                    'workerAgents': worker_agents,
+                },
+                'agentSelector': {'matchLabels': label_selector or {}}
+            }
+        )
+
+        if _does_string_contain_value(ssh_public_key):
+            self['agent']['sshPublicKey'] = ssh_public_key
+
+        if _does_string_contain_value(machine_cidr):
+            self['agent']['machineNetwork'] = [{'cidr': machine_cidr}],
+
+
+DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT = 60
+DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT = 300
+
+
+class ClusterDeployment(_BaseKubeAPIResource):
+    hive_api_group = 'hive.openshift.io'
+
+    def __init__(self, name: str, namespace: str = env_variables['namespace']):
+        _BaseKubeAPIResource.__init__(self, name, namespace)
+
+        self.crd_api = CustomObjectsApi(self.api_client)
+
+        self.__secret = None
+
+    @property
+    def secret(self):
+        return self.__secret
+
+    def create(
+            self,
+            platform: Platform,
+            install_strategy: InstallStrategy,
+            secret: Secret,
+            base_domain: str = env_variables['base_domain'],
+            **kwargs
+    ) -> None:
+
+        body = {
+            'apiVersion': f'{self.hive_api_group}/v1',
+            'kind': 'ClusterDeployment',
+            'metadata': self.ref,
+            'spec': {
+                'clusterName': self.ref['name'],
+                'baseDomain': base_domain,
+                'platform': platform,
+                'provisioning': {'installStrategy': install_strategy},
+                'pullSecretRef': secret.ref,
+            }
+        }
+        body['spec'].update(kwargs)
+        self.crd_api.create_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            body=body,
+            namespace=self.ref['namespace']
+        )
+        self.__secret = secret
+
+        logger.info('created cluster deployment %s: %s', self.ref, body)
+
+    def get(self) -> dict:
+        return self.crd_api.get_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            name=self.ref['name'],
+            namespace=self.ref['namespace']
+        )
+
+    def delete(self) -> None:
+        self.crd_api.delete_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            name=self.ref['name'],
+            namespace=self.ref['namespace']
+        )
+
+        logger.info('deleted cluster deployment %s', self.ref)
+
+    def status(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT
+    ) -> dict:
```

Comment:
Why `status` returns a `dict`?  I find this name misleading.

Reply:
it reflects a Status key in the crd that contains a dict

Reply:
Can you add a docstring?

Reply:
done